docs/developer/base/basics.md

```diff
@@ -1,3 +1,165 @@
 # Basics
 
 -----
+
+The AgentCheck base class in base.py contains the logic that all Checks inherit.
+
+In addition to the integrations inheriting from AgentCheck, other classes that inherit from AgentCheck include:
+
+- [PDHBaseCheck](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/checks/win/winpdh_base.py)
+- [OpenMetricsBaseCheck](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/checks/openmetrics/base_check.py)
+- [GenericPrometheusCheck](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/checks/prometheus/base_check.py)
+- [PrometheusCheck](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/checks/prometheus/prometheus_base.py)
+- [KubeLeaderElectionBaseCheck](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/checks/kube_leader/base_check.py)
+
+
+## Getting Started
+In each Check, Datadog looks for `__version__` and a subclass of AgentCheck at the root of every package.
+
+Below is an example of the `__init__.py` folder for AwesomeCheck:
+```python
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from .__about__ import __version__
+from .check import AwesomeCheck
+
+__all__ = ['__version__', 'AwesomeCheck']
+```
+
+Note that we look for `__version__` in `__about__.py` in this case, however it does not necessarily need to be put there. 
+
+This value is used in Datadog's check:
+
+```
+=========
+Collector
+=========
+
+  Running Checks
+  ==============
+
+    AwesomeCheck (0.0.1)
+    -------------------
+      Instance ID: 1234 [OK]
+      Configuration Source: file:/etc/datadog-agent/conf.d/awesomecheck.d/awesomecheck.yaml
+      Total Runs: 12
+      Metric Samples: Last Run: 242, Total: 2,904
+      Events: Last Run: 0, Total: 0
+      Service Checks: Last Run: 0, Total: 0
+      Average Execution Time : 49ms
+      Last Execution Date : 2020-10-26 19:09:22.000000 UTC
+      Last Successful Execution Date : 2020-10-26 19:09:22.000000 UTC
+
+...
+```
+
+## Checks
+AgentCheck contains functions that you use to execute checks and submit data to Datadog.
+
+include link to the app for each 
+
+### Metrics
+The list of what is collected from your system by each integration. More information on metrics can be found [here.](https://docs.datadoghq.com/developers/metrics/types/) You can find the metrics for each integration in that integrations metadata.csv file. You can also set up [custom metrics](https://docs.datadoghq.com/developers/metrics/), so if the integration doesn’t offer a metric out of the box, you can usually add it.
```

Comment:
```suggestion
This list enumerates what is collected from your system by each integration. For more information on metrics, see the [Metric Types documentation.](https://docs.datadoghq.com/developers/metrics/types/) You can find the metrics for each integration in that integration's `metadata.csv` file. You can also set up [custom metrics](https://docs.datadoghq.com/developers/metrics/), so if the integration doesn’t offer a metric out of the box, you can usually add it.
```

We like to have link text be descriptive, rather than `here`.